posts/2024-10-31-Project-Goals-Oct-Update.md

```diff
@@ -0,0 +1,271 @@
+---
+layout: post
+title: "October Project Goals Update"
+author: Niko Matsakis
+team: Leadership Council <https://www.rust-lang.org/governance/teams/leadership-council>
+---
+The Rust project is currently working towards a [slate of 26 project
+goals](https://rust-lang.github.io/rust-project-goals/2024h2/goals.html), with 3 of them designed as [Flagship
+Goals](https://rust-lang.github.io/rust-project-goals/2024h2/goals.html#flagship-goals). This post provides selected
+updates on our progress towards these goals (or, in some cases, lack thereof). The full details for any particular goal
+are available in its associated [tracking issue on the rust-project-goals
+repository](https://github.com/rust-lang/rust-project-goals/milestone/2).
+
+## Flagship goals
+
+<div style="display: flex;" class="mt2 mb3">
+    <div style="flex: auto;"><a href='https://github.com/rust-lang/rust-project-goals/issues/105'><strong>Bring the Async Rust experience closer to parity with sync Rust</strong></a></div>
+    <div style="flex: initial;"><progress value="15" max="37"></progress>
+</div>
+</div>
+
+The biggest elements of our goal are solving the ["send bound" problem][sbp] via [return-type notation][RTN] (RTN) and adding support for async closures. This month we made progress towards both. For RTN, @compiler-errors extended the return-type notation landed support for using RTN in self-types like `where Self::method(): Send`. He also authored a [blog post with a call for testing](https://blog.rust-lang.org/inside-rust/2024/09/26/rtn-call-for-testing.html) explaining what RTN is and how it works. For async closures, the lang team reached a preliminary consensus on the `async Fn` syntax, with the understanding that it will also include some "async type" syntax. This rationale was documented in [RFC #3710](https://github.com/rust-lang/rfcs/pull/3710), which is now open for feedback. The team held a [design meeting on Oct 23](https://hackmd.io/@rust-lang-team/ryxu3YLx1e) and @nikomatsakis will be updating the RFC with the concusions.
```

Comment:
```suggestion
The biggest elements of our goal are solving the ["send bound" problem][sbp] via [return-type notation][RTN] (RTN) and adding support for async closures. This month we made progress towards both. For RTN, @compiler-errors extended the return-type notation landed support for using RTN in self-types like `where Self::method(): Send`. He also authored a [blog post with a call for testing](https://blog.rust-lang.org/inside-rust/2024/09/26/rtn-call-for-testing.html) explaining what RTN is and how it works. For async closures, the lang team reached a preliminary consensus on the `async Fn` syntax, with the understanding that it will also include some "async type" syntax. This rationale was documented in [RFC #3710](https://github.com/rust-lang/rfcs/pull/3710), which is now open for feedback. The team held a [design meeting on Oct 23](https://hackmd.io/@rust-lang-team/ryxu3YLx1e) and @nikomatsakis will be updating the RFC with the conclusions.
```